src/v0/destinations/fb_custom_audience/recordTransform.js

```diff
@@ -235,6 +214,67 @@ async function processRecordInputs(groupedRecordInputs) {
   return finalResponse;
 }
 
+function processRecordInputsV1(groupedRecordInputs) {
+  const { destination } = groupedRecordInputs[0];
+  const { message } = groupedRecordInputs[0];
+  const { isHashRequired, disableFormat, type, subType, isRaw, audienceId, userSchema } =
+    destination.Config;
+
+  let operationAudienceId = audienceId;
+  let updatedUserSchema = userSchema;
+  if (isEventSentByVDMV1Flow(groupedRecordInputs[0])) {
+    const { objectType } = getDestinationExternalIDInfoForRetl(message, 'FB_CUSTOM_AUDIENCE');
+    operationAudienceId = objectType;
+    updatedUserSchema = getSchemaForEventMappedToDest(message);
+  }
+
+  return preparePayload(groupedRecordInputs, {
+    audienceId: operationAudienceId,
+    userSchema: updatedUserSchema,
+    isRaw,
+    type,
+    subType,
+    isHashRequired,
+    disableFormat,
+  });
+}
+
+const processRecordInputsV2 = (groupedRecordInputs) => {
+  const { connection, message } = groupedRecordInputs[0];
+  const { isHashRequired, disableFormat, type, subType, isRaw, audienceId } =
+    connection.config.destination;
+  // Ref: https://www.notion.so/rudderstacks/VDM-V2-Final-Config-and-Record-EventPayload-8cc80f3d88ad46c7bc43df4b87a0bbff
+  const identifiers = message?.identifiers;
+  let userSchema;
+  if (identifiers) {
+    userSchema = Object.keys(identifiers);
+  }
+  const events = groupedRecordInputs.map((record) => ({
+    ...record,
+    message: {
+      ...record.message,
+      fields: record.message.identifiers,
+    },
+  }));
+  return preparePayload(events, {
+    audienceId,
+    userSchema,
+    isRaw,
+    type,
+    subType,
+    isHashRequired,
+    disableFormat,
+  });
+};
+
+function processRecordInputs(groupedRecordInputs) {
+  const event = groupedRecordInputs[0];
+  if (isEventSentByVDMV2Flow(event)) {
```

Comment:
Need to support VDMV2 too for Eventstream - not required immediately

Reply:
Sure we will take that up once we get clarity on that